src/App.js

```diff
@@ -1,26 +1,29 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
 
-export class App extends Component {
-  state = {
-    movies: data,
-  };
+export const App = () => {
+  const [movies, setMovies] = useState(data);
+
+  const addMovie = (movie) => {
+    const hasSameMovie = movies
+      .some(film => film.imdbId === movie.imdbId);
```

Comment:
No need to create this duplicate. Just check if (movies.some....) and if there's already such movie in the list - do nothing (return) else - add it to the list.